package structDefinitions;

message Point2D {
    required double x = 1;
    required double y = 2;
}

message Point2DSet {
	repeated Point2D points = 1;
}

message Point3D {
	required double x = 1;
	required double y = 2;
	required double z = 3;
}

message Point3DSet  {
	repeated Point3D points = 1;
}

message Vertex {
	required Point3D point = 1;
	required double number = 2;
}

message VertexSet {
	repeated Vertex vertexes = 1;
}

message Edge {
	required Point3D v1 = 1;
	required Point3D v2 = 2;
}

message EdgeSet {
	repeated Edge edges = 1;
}

message TriangleFace {
	required Point3D v1 = 1;
	required Point3D v2 = 2;
	required Point3D v3 = 3;
}

message TriangleFaceSet {
	repeated TriangleFace triangleFaces = 1;
}

message Block {
	required Point3D v1 = 1;
	required Point3D v2 = 2;
	required Point3D v3 = 3;
	required Point3D v4 = 4;
}

message BlockSet {
	repeated Block blocks = 1;
}